tests/test_server.rs

```diff
@@ -476,6 +477,125 @@ async fn test_body_brotli() {
+            .wrap(Compress::new(ContentEncoding::Zstd))
+            .service(web::resource("/").route(web::to(move || HttpResponse::Ok().body(STR))))
+    });
+
+    // client request
+    let mut response = srv
+        .get("/")
+        .append_header((ACCEPT_ENCODING, "zstd"))
+        .no_decompress()
+        .send()
+        .await
+        .unwrap();
+    assert!(response.status().is_success());
+
+    // read response
+    let bytes = response.body().await.unwrap();
+
+    // decode
+    let mut e = ZstdDecoder::new(&bytes[..]).unwrap();
+    let mut dec = Vec::new();
+    e.read_to_end(&mut dec).unwrap();
+    assert_eq!(Bytes::from(dec), Bytes::from_static(STR.as_ref()));
+}
+
+#[actix_rt::test]
+async fn test_body_zstd_streaming() {
+    let srv = actix_test::start_with(actix_test::config().h1(), || {
+        App::new()
+            .wrap(Compress::new(ContentEncoding::Zstd))
+            .service(web::resource("/").route(web::to(move || {
+                HttpResponse::Ok()
+                    .streaming(TestBody::new(Bytes::from_static(STR.as_ref()), 24))
+            })))
+    });
+
+    // client request
+    let mut response = srv
+        .get("/")
+        .append_header((ACCEPT_ENCODING, "zstd"))
+        .no_decompress()
+        .send()
+        .await
+        .unwrap();
+    assert!(response.status().is_success());
+
+    // read response
+    let bytes = response.body().await.unwrap();
+
+    // decode
+    let mut e = ZstdDecoder::new(&bytes[..]).unwrap();
+    let mut dec = Vec::new();
+    e.read_to_end(&mut dec).unwrap();
+    assert_eq!(Bytes::from(dec), Bytes::from_static(STR.as_ref()));
+}
+
+#[actix_rt::test]
+async fn test_zstd_encoding() {
+    let srv = actix_test::start_with(actix_test::config().h1(), || {
+        App::new().service(
+            web::resource("/").route(web::to(move |body: Bytes| HttpResponse::Ok().body(body))),
+        )
+    });
+
+    let mut e = ZstdEncoder::new(Vec::new(), 5).unwrap();
+    e.write_all(STR.as_ref()).unwrap();
+    let enc = e.finish().unwrap();
+
+    // client request
+    let request = srv
+        .post("/")
+        .append_header((CONTENT_ENCODING, "zstd"))
+        .send_body(enc.clone());
+    let mut response = request.await.unwrap();
+    assert!(response.status().is_success());
+
+    // read response
+    let bytes = response.body().await.unwrap();
+    assert_eq!(bytes, Bytes::from_static(STR.as_ref()));
+}
+
+#[actix_rt::test]
+async fn test_zstd_encoding_large() {
+    let data = rand::thread_rng()
+        .sample_iter(&Alphanumeric)
+        .take(320_000)
+        .map(char::from)
+        .collect::<String>();
+
+    let srv = actix_test::start_with(actix_test::config().h1(), || {
+        App::new().service(
+            web::resource("/")
+                .app_data(web::PayloadConfig::new(320_000))
+                .route(web::to(move |body: Bytes| {
+                    HttpResponse::Ok().streaming(TestBody::new(body, 10240))
```

Comment:
to avoid any self-inversion bugs can we add an assertion of the Bytes content in here

Reply:
Sure ! Do you have any example 😄 ?
I have base this tests on `test_brotli_encoding_large` and `test_brotli_encoding`.
But they do not contains such asserts. I will be happy to add this asserts in this tests too.

Reply:
indeed they don't (yet) but this new test could lead the way

Reply:
actually i'm reading it wrong, the current asserts are fine